1.10.x+ Install fails with CURL_PROGRESSFUNC_CONTINUE was not declared when CPR_USE_SYSTEM_CURL=ON is used

### Description

Installation fails at `cmake build` with the error listed below on 2 separate Rocky 8 machines. 

I've tried to use 1.9.x but CPR_USE_SYSTEM_CURL doesn't appear to exist in that version, which provides the CMake files I need according to this https://github.com/offa/influxdb-cxx/issues/200#issuecomment-1587586390 

````
cmake --build .
Scanning dependencies of target cpr
[  3%] Building CXX object cpr/CMakeFiles/cpr.dir/accept_encoding.cpp.o
[  6%] Building CXX object cpr/CMakeFiles/cpr.dir/async.cpp.o
In file included from /opt/cpr/cpr/include/cpr/ssl_options.h:11,
                 from /opt/cpr/cpr/include/cpr/response.h:15,
                 from /opt/cpr/cpr/include/cpr/async_wrapper.h:8,
                 from /opt/cpr/cpr/include/cpr/async.h:4,
                 from /opt/cpr/cpr/cpr/async.cpp:1:
/opt/cpr/cpr/include/cpr/util.h: In function ‘int cpr::util::progressUserFunction(const T*, cpr::cpr_pf_arg_t, cpr::cpr_pf_arg_t, cpr::cpr_pf_arg_t, cpr::cpr_pf_arg_t)’:
/opt/cpr/cpr/include/cpr/util.h:26:36: error: ‘CURL_PROGRESSFUNC_CONTINUE’ was not declared in this scope
     static_assert(cancel_retval != CURL_PROGRESSFUNC_CONTINUE);
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/cpr/cpr/include/cpr/util.h:26:36: note: suggested alternative: ‘CURLOPT_PROGRESSFUNCTION’
     static_assert(cancel_retval != CURL_PROGRESSFUNC_CONTINUE);
                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~
                                    CURLOPT_PROGRESSFUNCTION
gmake[2]: *** [cpr/CMakeFiles/cpr.dir/build.make:96: cpr/CMakeFiles/cpr.dir/async.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:135: cpr/CMakeFiles/cpr.dir/all] Error 2
gmake: *** [Makefile:150: all] Error 2
````

### Example/How to Reproduce

1. `git clone https://github.com/libcpr/cpr.git` (also occurs with `--branch=1.10.x` as well)
2. `cd cpr && mkdir build && cd build`
3. `cmake .. -DCPR_USE_SYSTEM_CURL=ON`
4. `cmake --build .`

### Possible Fix

_No response_

### Where did you get it from?

GitHub (branch e.g. master)

### Additional Context/Your Environment

- OS: Rocky
- Version: 8.5
- Branch: both master and 1.10.x
- gcc 8.5.0, 12.2.0
- cmake 3.20.2, 3.17.2
- curl 7.61.1

--- a/include/cpr/util.h
+++ b/include/cpr/util.h
@@ -23,7 +23,9 @@ size_t writeUserFunction(char* ptr, size_t size, size_t nmemb, const WriteCallba
 template <typename T = ProgressCallback>
 int progressUserFunction(const T* progress, cpr_pf_arg_t dltotal, cpr_pf_arg_t dlnow, cpr_pf_arg_t ultotal, cpr_pf_arg_t ulnow) {
     const int cancel_retval{1};
+#ifdef CURL_PROGRESSFUNC_CONTINUE // Not always defined. Ref: https://github.com/libcpr/cpr/issues/932
     static_assert(cancel_retval != CURL_PROGRESSFUNC_CONTINUE);
+#endif // CURL_PROGRESSFUNC_CONTINUE
     return (*progress)(dltotal, dlnow, ultotal, ulnow) ? 0 : cancel_retval;
 }
 int debugUserFunction(CURL* handle, curl_infotype type, char* data, size_t size, const DebugCallback* debug);

